tests/test_batch_optimization.py

```
@@ -0,0 +1,146 @@
+    for case in edge_cases:
+        try:
+            batch_size = calculate_optimal_batch_size(case["cyclins"])
+            logger.info(f"{case['desc']} (cyclins={case['cyclins']}) → Batch size: {batch_size}")
+
+            # Additional validation for extreme values
+            if batch_size <= 0:
+                logger.error(f"❌ Invalid batch size: {batch_size} (must be positive)")
+            elif batch_size > 1024:
+                logger.warning(f"⚠️ Very large batch size: {batch_size} (may cause memory issues)")
+
+        except Exception as e:
```

Pylint (3.3.7)

[warning] Catching too general exception Exception

(W0718)